server/indexer_service.go

```diff
@@ -94,6 +94,13 @@ func (eis *EVMIndexerService) OnStart() error {
 	}
 	if lastBlock == -1 {
 		lastBlock = latestBlock
+	} else if lastBlock < status.SyncInfo.EarliestBlockHeight {
+		// to avoid infinite failed to fetch block error when lastBlock is smaller than earliest
+		lastBlock = status.SyncInfo.EarliestBlockHeight
```

Comment:
Does it work? What is the value of earliest block if a node prune in the middle.

Reply:
we might need skip the pruned blocks with retries

Reply:
earliest could be middle block, I could reproduce with config like
```
"min-retain-blocks": 10,
"pruning-interval": 10,
"pruning-keep-recent": 4,
"state-sync": {
    "snapshot-interval": 1
}
```